docs/pages/application-access/guides/azure.mdx

```diff
@@ -0,0 +1,498 @@
+
+A user can omit the `--azure-identity` flag if they are only authorized to
+access a single Azure identity, but otherwise an empty `--azure-identity` will
+result in an error.
+
+If the command succeeds, you will see information about the user's chosen Azure
+identity similar to the following:
+
+```text
+[
+  {
+    "environmentName": "AzureCloud",
+    "homeTenantId": "00000000-0000-0000-0000-000000000000",
+    "id": "00000000-0000-0000-0000-000000000000",
+    "isDefault": true,
+    "managedByTenants": [],
+    "name": "Microsoft Azure Sponsorship",
+    "state": "Enabled",
+    "tenantId": "00000000-0000-0000-0000-000000000000",
+    "user": {
+      "assignedIdentityInfo": "MSIResource-/subscriptions/0000000000000-0000-0000-000000000000/resourceGroups/my-resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/teleport-azure",
+      "name": "userAssignedIdentity",
+      "type": "servicePrincipal"
+    }
+  }
+]
+
+Logged into Azure app "azure-cli".
+Your identity: /subscriptions/0000000000000-0000-0000-000000000000/resourceGroups/my-resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/teleport-azure
+Example Azure CLI command: tsh az vm list
+```
+
+### Execute Azure CLI commands
+
+At this point, you can run `az` commands using the Teleport Application Service
+by prefixing them with `tsh`. To list VMs running in your Azure resource group,
+for example, run the following command:
+
+```code
+$ tsh az vm list
+```
+
+<Notice type="tip">
+
+If you're not seeing the expected VMs at this point, double-check that your
+Azure managed identity is assigned the "Reader" role at the scope of your
+resource group.
+
+</Notice>
+
+### Use Azure CLI applications without `tsh`
+
+In addition to running `az` commands via `tsh`, you can grant secure access to
+any CLI application that executes commands against Azure's APIs.
+
+To do this, use `tsh` to start a local proxy that forwards traffic from your CLI
+application to the Teleport Application Service. The Application Service uses an
+Azure managed identity to fetch an authentication token from Azure, which your
+CLI application uses to authenticate requests to Azure's APIs.
+
+To start the local proxy, run the following `tsh` command:
+
+```code
+$ tsh proxy azure 
+```
+
+The command will print the address of the local proxy server along with `export`
+commands for assigning environment variables. Azure CLI applications read these
+variables in order to request an authentication token for Azure's APIs:
+
+```text
+Started Azure proxy on http://127.0.0.1:54330.
+To avoid port randomization, you can choose the listening port using the --port flag.
+
+Use the following credentials and HTTPS proxy setting to connect to the proxy:
+
+  export AZURE_CONFIG_DIR=/Users/myuser/.azure-teleport
+  export HTTPS_PROXY=http://127.0.0.1:54330
+  export HTTP_PROXY=http://127.0.0.1:54330
+  export MSI_ENDPOINT=https://azure-msi.teleport.dev
```

Comment:
This is being discussed in the code PR, but most likely, we will have some auth mechanism for accessing `MSI_ENDPOINT`: https://github.com/gravitational/teleport/pull/18675#discussion_r1040641188. 

The likely solution is a random hex attached at the end:
```
export MSI_ENDPOINT=https://azure-msi.teleport.dev/eedfd5b55257c0aaa58f
```

But also with an option to set this by the user:
```
$ export MSI_ENDPOINT=https://azure-msi.teleport.dev/MY-CUSTOM-SECRET
$ tsh proxy azure --port 3500
Started Azure proxy on http://127.0.0.1:3500.

Use the following credentials and HTTPS proxy setting to connect to the proxy:

  export AZURE_CONFIG_DIR=/Users/tener/.azure-teleport-azure
  export HTTPS_PROXY=http://127.0.0.1:3500
  export MSI_ENDPOINT=https://azure-msi.teleport.dev/MY-CUSTOM-SECRET
  export REQUESTS_CA_BUNDLE=/Users/tener/.tsh/keys/boson.tener.io/tener-app/boson.tener.io/azure-localca.pem
```

This may be one of the details that will have to wait until the code PR is merged to master, though.